pygmt/src/grdcontour.py

```diff
@@ -96,32 +100,44 @@ def grdcontour(self, grid, **kwargs):
     Example
     -------
     >>> import pygmt
-    >>> # load the 15 arc-minutes grid with "gridline" registration
-    >>> # in a specified region
+    >>> # Load the 15 arc-minutes grid with "gridline" registration in the
+    >>> # specified region
     >>> grid = pygmt.datasets.load_earth_relief(
     ...     resolution="15m",
     ...     region=[-92.5, -82.5, -3, 7],
     ...     registration="gridline",
     ... )
-    >>> # create a new plot with pygmt.Figure()
+    >>> # Create a new plot with pygmt.Figure()
     >>> fig = pygmt.Figure()
-    >>> # create the contour plot
+    >>> # Create the contour plot
     >>> fig.grdcontour(
-    ...     # pass in the grid downloaded above
+    ...     # Pass in the grid downloaded above
     ...     grid=grid,
-    ...     # set the interval for contour lines at 250 meters
+    ...     # Set the interval for contour lines at 250 meters
     ...     interval=250,
-    ...     # set the interval for annotated contour lines at 1,000 meters
+    ...     # Set the interval for annotated contour lines at 1,000 meters
     ...     annotation=1000,
-    ...     # add a frame for the plot
+    ...     # Add a frame for the plot
     ...     frame="a",
-    ...     # set the projection to Mercator for the 10 cm figure
+    ...     # Set the projection to Mercator for the 10 cm figure
     ...     projection="M10c",
     ... )
-    >>> # show the plot
+    >>> # Show the plot
     >>> fig.show()
     """
     kwargs = self._preprocess(**kwargs)
```

Comment:
Add some codes for backward-compatibility:
```suggestion
    kwargs = self._preprocess(**kwargs)
    
    # Backward compatibility with the old syntax like (100, "e", "f10p", "gred").
    if is_nonstr_iter(kwargs.get("A")) and any(
        i[0] in "acdefgijlLnoprtuvwx=" for i in kwargs("A") if isinstance(i, str)
    ):
        msg = (
            "Argument of the parameter 'annotation'/'A' is using the old, deprecated "
            "syntax. Please refer to the PyGMT documentation for the new syntax. "
            "The warning will be removed in v0.14.0 and the old syntax will no longer "
            "be supported. "
        )
        warnings.warn(msg, catalog=FutureWarning, stacklevel=2)
        kwargs["A"] = "+".join(f"{item}" for item in kwargs["A"])
```